### PR TITLE
Trying to fix issue #145 here.

### PR DESCRIPTION
**Pull request by R-Larocque**
Trying to fix issue #145 here.

## Summary by Sourcery

Refine Windows flashing, formatting, and USB detection flows while tightening error handling and expanding test coverage across the app.

Bug Fixes:
- Fix Windows ISO flashing to fall back to direct copy when install.wim/esd is missing and to correctly handle large WIMs on FAT32 using wimlib splitting.
- Ensure unmount/remount, formatting, and flashing paths return False instead of propagating subprocess and OS errors, including missing tools (dd, mkfs, wimlib, badblocks, 7z).
- Prevent crashes and misbehaviour in USB/device handling by guarding empty device nodes, normalising mount paths, and safely resolving raw block devices.
- Avoid theme and state inconsistencies by using the unified AppState instance (`state`) for theme persistence and environment propagation.
- Tighten device path validation for Windows flashing to reject invalid targets and avoid corrupting non-disk paths.

Enhancements:
- Introduce helpers for mounting ISOs, creating partitions, and locating UEFI NTFS images to centralise platform-specific logic.
- Improve Windows boot media creation by copying EFI/boot files more robustly and injecting optional registry tweaks for hardware and account bypass via wim tools.
- Make log output, status messages, and accessibility labels more consistent and informative across GUI operations and background workers.
- Rename the main window class to `LufusWindow` and clean up GUI/theme handling to better support multiple themes and user configuration.
- Refactor multiple modules for style consistency, clearer constants, and safer defaults (e.g. volume label sanitisation, large disk warnings).
- Remove unused imports, dead code, and a duplicate pyproject [project] section to simplify the build and codebase.

Tests:
- Add and extend tests around unmount/remount behaviour, error paths, and subprocess failures to guarantee graceful handling and callbacks.
- Expand regression tests for Windows flashing, ISO detection, Ventoy/GRUB installation, USB discovery, and state typing to lock in the new behaviour.
- Strengthen get_usb_info, find_usb, and formatting tests to cover timeout handling, label fallbacks, partition parsing, and block size probing edge cases.

### CODE REVIEW

**Comment by Radiump123** [comment]
@sourcery-ai review

**Comment by Radiump123** [comment]
@sourcery-ai review

**Comment by Radiump123** [comment]
@sourcery-ai review

**Comment by Radiump123** [comment]
@sourcery-ai review

**Comment by Radiump123** [comment]
@sourcery-ai review

File: src/lufus/writing/flash_usb.py [hunk 1]
```diff
@@ -75,14 +76,10 @@ def _status(msg: str) -> None:
...
-        _status(
-            f"Writing {iso_size:,} bytes to {device}, this may take several minutes..."
-        )
+        _status(f"Writing {iso_size:,} bytes to {shlex.quote(device)}, this may take several minutes...")
 
         try:
-            process = subprocess.Popen(
-                dd_args, stderr=subprocess.PIPE, stdout=subprocess.DEVNULL
-            )
+            process = subprocess.Popen(dd_args, stderr=subprocess.PIPE, stdout=subprocess.DEVNULL)
```
[the remark this answers is not in the document]
Comment (R-Larocque) [reply src/lufus/writing/flash_usb.py:96]: > **security (python.lang.security.audit.dangerous-subprocess-use-audit):** Detected subprocess function 'Popen' without a static string. If this data can be controlled by a malicious actor, it may be an instance of command injection. Audit the use of this call to ensure it is not controllable by an external resource. You may consider using 'shlex.escape()'.
> 
> _Source: opengrep_

Can't be fixed easily without massive rewrite, won't fix.

**Comment by Radiump123** [comment]
still got bugs to fix and stuff to test, including bugs in flashing logic...

**Comment by Radiump123** [comment]
wonderful, it starts, now i gotta actually  test it for bad bugs

**Comment by Radiump123** [comment]
tested, flashing is broken, i will move back pkexec at the start temporarily until i find a better solution

**Comment by Radiump123** [comment]
ok im gonna undo some of my changes i broke shit ig

**Comment by Radiump123** [comment]
Nice, sourcery gave up on me. Gonna audit everything myself.

**Comment by Radiump123** [comment]
i'm fixing security issues rn

**Comment by Radiump123** [comment]
wow i broke smth again
lets unbreak it
[ahh](https://i.imgflip.com/4emb6h.mp4)

**Comment by Radiump123** [comment]
ok it wotn flash nvme cuz goofy ahh bugs
lets fix that too

**Comment by Radiump123** [comment]
problems left:
- some minor bugs
- status bar is goofy
- no nuitka (i just love nuitka)
- some stuff i just dont like but hey it works
- some stuff to test
- idk about settings persistence
- nothing easy for making themes
- should add more languages and enhance existing translations, and translate button alt text
- A LOT of deps
- Etc.


but for this pr, ill just finish fixing bugs and open it

**Comment by R-Larocque** [comment]
ok it seems good. opening pr...

**Comment by Splayer0** [comment]
LGTM will merge, test and reopen if needed
